arangod/RestServer/DatabaseFeature.cpp

```diff
@@ -1269,7 +1266,13 @@ int DatabaseFeature::iterateDatabases(VPackSlice const& databases) {
       // open the database and scan collections in it
 
       // try to open this database
-      auto* database = engine->openDatabase(it, _upgrade).release();
+      arangodb::CreateDatabaseInfo info(server());
+      info.allowSystemDB(true);
+      auto res = info.load(it, VPackSlice::emptyArraySlice());
+      if(res.fail()){
+        THROW_ARANGO_EXCEPTION(res);
+      }
+      auto* database = engine->openDatabase(std::move(info), _upgrade).release();
```

Comment:
Can we change this so an `std::unique_ptr` takes ownership here (locally), and releases ownership after the insert into `newLists->_databases` succeeds.